core/src/main/java/org/elasticsearch/cluster/routing/allocation/command/AllocateEmptyPrimaryAllocationCommand.java

```diff
@@ -127,6 +129,10 @@ public RerouteExplanation execute(RoutingAllocation allocation, boolean explain)
                 shardRouting.unassignedInfo().getLastAllocationStatus());
         }
 
+        Logger logger = Loggers.getLogger(AllocateEmptyPrimaryAllocationCommand.class);
+        logger.warn("Allocating an empty primary for [{}][{}] on node [{}]. This action can cause data loss. If the " +
+            "old primary rejoins the cluster, its copy of this shard will be deleted.", index, shardId, node);
```

Comment:
I added the "If the old primary rejoins..." because I thought it was helpful. This does seem like the kind of place where we may forget to update it if we change the commands to work differently, so I could see leaving it out too.